### ISSUE TITLE
Allow to open a copy of decompiled Minecraft class for editing (to be able to rename local variables)

### ISSUE THREAD

**Issue by Barteks2x** [minecraft-dev/MinecraftDev#218]
The one problem I have with understanding parts of MC code is that local variable names are meaningless, and I have to keep track of them. This is very annoying and slows me down a lot. It would be a lot nicer if I could edit them directly in decompiled code, without copying it to my own code and wasting 10 minutes trying to make it compile.

**Comment by DenWav** [comment]
I assume you're meaning to open it as a scratch (virtual) file, rather than something which would actually be in the project?

**Comment by Niko-sk2x** [comment]
Yes, something like that. Ideally editing the file directly would be possible, but that isn't something I would expect to be doable from a plugin.

Ideally that edited version could be saved so that I wouldn't easily loose it.

As I said, the only purpose of this would be to be able to rename variables inside these classes, if this is possible to do in any other way - it would be fine too.

Adding these files to my project manually is possible, but it quickly gets messy.

**Comment by DenWav** [comment]
The sources are retrieved from the jar provided by `setupDecW`, so don't expect to start editing them any time soon. Best option would probably be to just cache the changes in the `.idea` directory so they are persisted, while not cluttering the project.

**Comment by Niko-sk2x** [comment]
I know editing them directly would be an issue, so that's why I just want to be able to quickly make a copy for editing, that actually compiles and has full autocomplete support.

**Comment by Bluexin** [comment]
How about adding them to a local mappings file? wouldn't that work?

**Comment by Niko-sk2x** [comment]
There are no mappings for LVT. And local variables is what I want to rename. Sometimes a bit of code rearrangement and comments are also useful (for example using early return instead of `if`)

**Comment by DenWav** [comment]
I'm still not quite sure if I totally understand how this should be implemented in any way that would actually be useful.

**Comment by DenWav** [comment]
Coming back to this I think I have an idea of how this could be done in a useful way. Don't know when I'll get to this, development on the plugin has slowed down a lot lately. Not sure when it'll pick up again.
